drivers/frequency/admfm2000/admfm2000.c

```diff
@@ -0,0 +1,217 @@
+/***************************************************************************//**
+ *   @file   admfm2000.c
+ *   @brief  Implementation of admfm2000 Driver.
+ *   @author Ramona Nechita (ramona.nechita@analog.com)
+********************************************************************************
+ * Copyright 2024(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "admfm2000.h"
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+#include "no_os_print_log.h"
+
+
+int32_t admfm2000_get_gain(struct admfm2000_dev *dev, uint8_t chan,
+			   int32_t *gain)
+{
+	int32_t ret;
+	uint8_t i;
+	uint8_t bit;
+	int32_t tmp = 0;
+
+	if (chan > 1)
+		return -EINVAL;
+
+	if (chan)
+		for (i = 0; i < ADMFM2000_DSA_GPIOS; i++) {
+			ret = no_os_gpio_get_value(dev->gpio_dsa1[i], &bit);
+			if (ret != 0)
+				return ret;
+			tmp |= (bit << i);
+		} else
```

Comment:
the indentation is wrong here and so are the curly braces, the else is misplaced, you gotta review this portion and redo it

Reply:
also you could store the gpio_dsa in an 2d array and you could have easier indexing like this:
```
dev->gpio_dsa[0][i] - chan 0
dev->gpio_dsa[1][i] - chan 1
```